app/models/document.rb

```diff
@@ -52,6 +55,11 @@ class Document < ApplicationRecord
 
   scope :with_tag, ->(tag) { where("tags @> ?", "\"#{tag}\"") }
 
+  before_create do
+    self.api_user ||= Current.api_user
+    self.user ||= Current.user
```

Comment:
Will this always be set because we do `before_action :set_current_user` in the application_controller ?

Reply:
no this is to assign the user to the document object - it won't always be set since in the api controller we won't have a user logged in necessarily - you can, but you can also log out and send the application with token and that's when api user gets set